compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -550,6 +551,27 @@ fn type_as_slice(
     })
 }
 
+// fn as_struct(self) -> Option<(StructDefinition, [Type])>
+fn type_as_struct(
+    arguments: Vec<(Value, Location)>,
+    return_type: Type,
+    location: Location,
+) -> IResult<Value> {
+    type_as(arguments, return_type, location, |typ| {
+        if let Type::Struct(struct_type, generics) = typ {
+            Some(Value::Tuple(vec![
+                Value::StructDefinition(struct_type.borrow().id),
+                Value::Slice(
+                    generics.into_iter().map(Value::Type).collect(),
+                    Type::Quoted(QuotedType::Type),
```

Comment:
This should be the slice type, not the element type

Reply:
Done!